pkg/operator/operator.go

```diff
@@ -161,16 +180,93 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		syncErrrorMetric.WithLabelValues("SyncError").Set(0)
 	}
 
-	if _, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(availableCnd),
-	); updateErr != nil {
+	updateFuncs := []v1helpers.UpdateStatusFunc{}
+	updateFuncs = append(updateFuncs, v1helpers.UpdateConditionFn(availableCnd))
+	allowUpgradeCond := operatorapi.OperatorCondition{
+		Type:   controllerName + operatorapi.OperatorStatusTypeUpgradeable,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if blockUpgrade {
+		blockUpgradeMessage := fmt.Sprintf("Marking cluster un-upgradeable because %s", blockUpgradeReason)
+		klog.Infof(blockUpgradeMessage)
+		c.eventRecorder.Warningf("VsphereOlderVersonDetected", "Marking cluster un-upgradeable because %s", blockUpgradeReason)
```

Comment:
Sorry, I made typo in VsphereOlder**Verson**Detected in my previous comment. Can you please fix it?